thetis/options.py

```diff
@@ -726,16 +751,27 @@ def add_tracer_2d(self, label, name, filename, shortname=None, unit='-', source=
         :kwarg shortname: short version of name, e.g. 'Tracer'
         :kwarg unit: units for field, e.g. '-'
         :kwarg source: associated source term
+        :kwarg diffusivity: associated diffusivity coefficient
         """
+        assert isinstance(label, str)
+        assert isinstance(name, str)
+        assert isinstance(filename, str)
+        assert shortname is None or isinstance(shortname, str)
+        assert isinstance(unit, str)
+        if label in self.tracer_metadata:
+            print_output(f"Field '{label}' already exists. It will be overwritten.")
         assert ' ' not in label, "Labels cannot contain spaces"
         assert ' ' not in filename, "Filenames cannot contain spaces"
-        self.tracer_metadata[label] = {
+        options = TracerFieldOptions()
+        options.source = source
+        options.diffusivity = diffusivity
+        self.tracer_metadata[label] = AttrDict({
             'name': name,
             'shortname': shortname or name,
             'unit': unit,
             'filename': filename,
-            'source': source,  # TODO: Validate input
-        }
+            'options': options,
```

Comment:
So now we can access tracer options like this
```python
options.tracer_metadata[label].name
options.tracer_metadata[label].options.diffusivity
```
maybe this would be more logical
```python
options.tracer[label].metadata.name
options.tracer[label].diffusivity
```
i.e. the fields are not inside the metadata dict?

Reply:
Good idea, makes sense